### PR TITLE
make verify a blocking build

### PR DESCRIPTION
**Pull request by mikedanese**
cc @lavalamp

### CODE REVIEW

File: mungegithub/mungers/submit-queue.go [hunk 1]
```diff
@@ -51,11 +51,12 @@ const (
...
-	jenkinsE2EContext  = "Jenkins GCE e2e"
-	jenkinsUnitContext = "Jenkins unit/integration"
-	jenkinsNodeContext = "Jenkins GCE Node e2e"
-	travisContext      = "continuous-integration/travis-ci/pr"
-	sqContext          = "Submit Queue"
+	jenkinsE2EContext    = "Jenkins GCE e2e"
+	jenkinsUnitContext   = "Jenkins unit/integration"
+	jenkinsVerifyContext = "Jenkins verification"
+	jenkinsNodeContext   = "Jenkins GCE Node e2e"
+	travisContext        = "continuous-integration/travis-ci/pr"
```
Comment (spxtr) [inline mungegithub/mungers/submit-queue.go]: Can remove this while you're at it, but it's not a big deal.

Comment (mikedanese) [inline mungegithub/mungers/submit-queue.go]: done.

**Comment by eparis** [comment]
the 'k8s-bot test this' comment will cause this to be re-tested?  Does this (and node e2e) need to be added to the ping-stale* mungers?

File: mungegithub/mungers/submit-queue.go [hunk 2]
```diff
@@ -466,7 +467,7 @@ func (sq *SubmitQueue) AddFlags(cmd *cobra.Command, config *github.Config) {
 		[]string{},
 		"Comma separated list of jobs in Jenkins to use for stability testing that needs only weak success")
 	cmd.Flags().StringSliceVar(&sq.RequiredStatusContexts, "required-contexts", []string{}, "Comma separate list of status contexts required for a PR to be considered ok to merge")
-	cmd.Flags().StringSliceVar(&sq.RequiredRetestContexts, "required-retest-contexts", []string{jenkinsE2EContext, jenkinsUnitContext, jenkinsNodeContext}, "Comma separate list of statuses which will be retested and which must come back green after the `retest-body` comment is posted to a PR")
+	cmd.Flags().StringSliceVar(&sq.RequiredRetestContexts, "required-retest-contexts", []string{jenkinsE2EContext, jenkinsUnitContext, jenkinsNodeContext, jenkinsVerifyContext}, "Comma separate list of statuses which will be retested and which must come back green after the `retest-body` comment is posted to a PR")
```
Comment (lavalamp) [inline mungegithub/mungers/submit-queue.go]: Can you put these items on separate lines for readability?

File: mungegithub/mungers/submit-queue.go [hunk 3]
```diff
@@ -51,11 +51,11 @@ const (
...
 	claHumanLabel          = "cla: human-approved"
 
-	jenkinsE2EContext  = "Jenkins GCE e2e"
-	jenkinsUnitContext = "Jenkins unit/integration"
-	jenkinsNodeContext = "Jenkins GCE Node e2e"
-	travisContext      = "continuous-integration/travis-ci/pr"
-	sqContext          = "Submit Queue"
+	jenkinsE2EContext    = "Jenkins GCE e2e"
+	jenkinsUnitContext   = "Jenkins unit/integration"
+	jenkinsVerifyContext = "Jenkins verification"
```
Comment (lavalamp) [inline mungegithub/mungers/submit-queue.go]: Is there a reason to have these constants listed here? @eparis

Comment (eparis) [inline mungegithub/mungers/submit-queue.go]: they are hard coded in the 'ping-*' mungers I think.

Comment (mikedanese) [inline mungegithub/mungers/submit-queue.go]: Should they all be added to the stale ping checks? Currently jenkinsNodeContext is missing from that list.

Comment (eparis) [inline mungegithub/mungers/submit-queue.go]: I think they should be, but doesn't have to be this PR.

Comment (eparis) [inline mungegithub/mungers/submit-queue.go]: actually maybe it does need to be. I think many/most things in the queue have never run verification, right? so they'll just fall out of the queue and won't ever come back...

Comment (mikedanese) [inline mungegithub/mungers/submit-queue.go]: Done.

**Comment by mikedanese** [comment]
Ok tried to do a minimal cleanup. The required builds should really be plumbed through to the mungers eventually though.

File: mungegithub/mungers/submit-queue.go [hunk 4]
```diff
@@ -67,6 +67,13 @@ var (
 	_ = fmt.Print
 	// This MUST cause a RETEST of everything in the sq.RequiredRetestContexts
 	retestBody = fmt.Sprintf("@%s test this [submit-queue is verifying that this PR is safe to merge]", jenkinsBotName)
+
+	requiredContexts = []string{
```
Comment (eparis) [inline mungegithub/mungers/submit-queue.go]: Can you call this 'requiredRetestContexts' ?

Comment (eparis) [inline mungegithub/mungers/submit-queue.go]: Actually this is fine, i'm just a little worried/uncomfortable about the usage in the 'required-retest-contexts' flag...  I want that part broken out somehow...

**Comment by eparis** [comment]
@mikedanese @ixdy is the infrastructure going to be able to handle it when we hit it with 50+ 'verification' tests at once?  Or do we need to run some custom stale munger by hand over the next couple days to get caught up?

File: mungegithub/mungers/submit-queue.go (hunk 4 above)
Comment (lavalamp) [inline mungegithub/mungers/submit-queue.go]: we need to make these all flags somehow eventually if we want to run this thing against other repos.

**Comment by lavalamp** [comment]
LGTM

**Comment by mikedanese** [comment]
@eparis We can watch it. I think it will be ok to do it and let the queue drain.

**Comment by spxtr** [comment]
If you think there's going to be a huge rush of verify requests then I might suggest merging either early in the morning or late at night when PR Jenkins isn't busy.

**Comment by eparis** [comment]
when it triggers a retest I believe it triggers `test this` which is actually all 4!

**Comment by lavalamp** [comment]
Yeah, we've *got* to fix that...

**Pull request merged**